### PR TITLE
Update lint packages

### PR DESCRIPTION
**Pull request by lazytype**
Updates linter packages and fixes arising issues

## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/menu/src/Menu.tsx
Comment (brookescarlett) [inline packages/menu/src/Menu.tsx:115]: This PR is reminding me that I should take a look at focus-handling in the Menu component. I think we misunderstood some guidelines, and we can probably make this entire component dramatically simpler. (out of scope here, but more a note to self 🤔 )

Comment (brookescarlett) [reply packages/menu/src/Menu.tsx:115]: That said, these changes do require a changeset before merging in

File: .eslintrc.js
Comment (brookescarlett) [inline .eslintrc.js:30]: Do we need to leave the double comment here or can we just rm the line entirely

**Review by brookescarlett — APPROVED** [review]
This looks good to me, just be sure to add a Menu changeset before merging

File: packages/icon/src/Icon.spec.tsx
Comment (brookescarlett) [inline packages/icon/src/Icon.spec.tsx:48]: I think because you left the rule commented out this is no longer necessary

File: packages/icon/src/Icon.spec.tsx
Comment (brookescarlett) [inline packages/icon/src/Icon.spec.tsx:188]: ditto here

File: packages/menu/src/FocusableMenuItem.tsx
Comment (brookescarlett) [inline packages/menu/src/FocusableMenuItem.tsx:11]: Isn't this being done implicitly?

File: .eslintrc.js
Comment (lazytype) [reply .eslintrc.js:30]: 👍

File: packages/menu/src/FocusableMenuItem.tsx
Comment (lazytype) [reply packages/menu/src/FocusableMenuItem.tsx:11]: 👍

**Pull request merged**
